osu.Game.Tests/Visual/Multiplayer/TestSceneMultiSpectatorScreen.cs

```diff
@@ -314,6 +316,36 @@ public void TestSpectatingDuringGameplayWithLateFrames()
             AddUntilStep("player 2 playing from correct point in time", () => getPlayer(PLAYER_2_ID).ChildrenOfType<DrawableRuleset>().Single().FrameStableClock.CurrentTime > 30000);
         }
 
+        [Test]
+        public void TestPlayersLeaveWhileSpectating()
+        {
+            start(getPlayerIds(4));
+            sendFrames(getPlayerIds(4), 300);
+
+            loadSpectateScreen();
+
+            for (int count = 3; count >= 0; count--)
```

Comment:
it's test code so I won't pick on it too much, but the indices are recomputed 3 times in 6 lines, the last time even manually instead of calling the same method again